framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -191,6 +191,44 @@ void MediaPlayerImpl::preparePlayer(player_result_t &ret)
 	return notifySync();
 }
 
+player_result_t MediaPlayerImpl::prepareAsync()
+{
+	player_result_t ret = PLAYER_OK;
+
+	std::unique_lock<std::mutex> lock(mCmdMtx);
+	medvdbg("MediaPlayer prepare\n");
+
+	PlayerWorker &mpw = PlayerWorker::getWorker();
+	if (!mpw.isAlive()) {
+		meddbg("PlayerWorker is not alive\n");
+		return PLAYER_ERROR_NOT_ALIVE;
+	}
+
+	mpw.enQueue(&MediaPlayerImpl::prepareAsyncPlayer, shared_from_this(), std::ref(ret));
+	return ret;
+}
+
+void MediaPlayerImpl::prepareAsyncPlayer(player_result_t &ret)
+{
+	LOG_STATE_INFO(mCurState);
+
+	if (mCurState != PLAYER_STATE_CONFIGURED) {
+		meddbg("%s Fail : invalid state\n", __func__);
+		LOG_STATE_DEBUG(mCurState);
+		ret = PLAYER_ERROR_INVALID_STATE;
+		return;
```

Comment:
you should add handling for error status to notifyobserver.

Reply:
I think it is ok, not to throw it to notifyObserver.
Because prepareAsync returned false.
So the user will not wait for the observer on this.

Reply:
I see, it's an asynchronous api, we can not get correct `ret` value from prepareAsyncPlayer() immediately.
I will fix it.

Reply:
Fixed.